docs/options.md

```diff
@@ -299,6 +299,30 @@ the package is compatible with all versions of Python that it can build.
 CIBW_PROJECT_REQUIRES_PYTHON: ">=3.6"
 ```
 
+###  `CIBW_PRERELEASE_PYTHONS` {: #prerelease-pythons}
+> Enable building with pre-release versions of Python
+
+By default, cibuildwheel only builds wheels compatible with minor versions of
+Python. However, during the Beta period when new versions of Python are being
+tested, cibuildwheel will often gain early support for beta releases. If you
+would like to test wheel building with these versions, you can enable this
+flag.  Please _do not_ upload these wheels to PyPI, as they are not guaranteed
+to work with the final Python release. Once Python is ABI stable and enters the
+release candidate phase, that version of Python will become available without
+this flag.
```

Comment:
Nice job with this, it looks good.

A couple notes-

- The first sentence is a little confusing "compatible with minor versions of Python" Actually do we need the first sentence at all? We could just start with "During the beta period when new versions of Python are being tested..."
- Adding an admonition would be nice here, to draw attention to the ABI compatibility note. Perhaps:

  ```
  !!! caution
      Please _do not_ upload these wheels to PyPI, as they are not guaranteed... 
  ```

This is my suggestion, if you want to copy/paste:

```
During the beta period, when new versions of Python are being tested,
cibuildwheel will often gain early support for beta releases. If you would
like to test wheel building with these versions, you can enable this flag.

!!! caution
    Please _do not_ upload these wheels to PyPI, as they are not guaranteed to
    work with the final Python release. Once Python is ABI stable and enters
    the release candidate phase, that version of Python will become available
    without this flag.
```

Reply:
I copy/pasted @joerick suggestion.
I think I understand the intent of @henryiii and that's still missing from the docs:
"you won't get pre-release for 3.7.y, 3.8.y, 3.9.y. You'll only get pre-release for a minor version 3.x if 3.x.0 isn't there yet"
If that was the intent, I don't really know  how to phrase that properly (and given it acts on non ABI stable versions, "3.x.0" isn't right either)

Reply:
@henryiii, did I understand correctly your intent ? If so, do you or @joerick have something to propose ?

Reply:
Yes, I understand. But it's quite hard to write... I also think that anyone who might want to use this option would probably understand that they don't care about the ABI-compatible patch betas. Given the warning below, it should be clear that we're talking about the ABI-incompatible ones, I think?

Reply:
I think that's fine.

Reply:
ok, let's stick with this then.